### ISSUE TITLE
Can't break loop when using > 1 worker with DataLoader

### ISSUE THREAD

**Issue by ThomasDelteil** [apache/incubator-mxnet#13126]
## Description

When using more than 1 worker with a DataLoader and starting the iteration and breaking the loop straight away, the worker crash and throw an error.

## Environment info (Required)

```
MXNet 1.3.1
```

## Minimum reproducible example

```python

train_dataset = mx.gluon.data.vision.MNIST(train=True).transform_first(mx.gluon.data.vision.transforms.ToTensor())
train_data = mx.gluon.data.DataLoader(train_dataset, shuffle=True, last_batch='rollover', batch_size=100, num_workers=2)

for data, label in train_data:
    break
```
```
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/ubuntu/anaconda3/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/mxnet/gluon/data/dataloader.py", line 195, in fetcher_loop
    idx, batch = data_queue.get()
  File "/home/ubuntu/anaconda3/lib/python3.6/multiprocessing/queues.py", line 337, in get
    return _ForkingPickler.loads(res)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/mxnet/gluon/data/dataloader.py", line 57, in rebuild_ndarray
    fd = fd.detach()
  File "/home/ubuntu/anaconda3/lib/python3.6/multiprocessing/resource_sharer.py", line 58, in detach
    return reduction.recv_handle(conn)
  File "/home/ubuntu/anaconda3/lib/python3.6/multiprocessing/reduction.py", line 182, in recv_handle
    return recvfds(s, 1)[0]
  File "/home/ubuntu/anaconda3/lib/python3.6/multiprocessing/reduction.py", line 153, in recvfds
    msg, ancdata, flags, addr = sock.recvmsg(1, socket.CMSG_LEN(bytes_size))
ConnectionResetError: [Errno 104] Connection reset by peer

```

**Comment by leleamol** [comment]
@ThomasDelteil Thanks for submitting this issue. I am labeling this issue so that MXNet Community members can help resolve it.

@mxnet-lebel-bot [Gluon, Data-loading]

**Comment by harshp8l** [comment]
@mxnet-label-bot [Gluon, Data-loading]

**Comment by abhinavs95** [comment]
Hi @ThomasDelteil I think this issue is fixed in 1.4.0, I am able to run your example with no errors. Also mentioned in #14541
